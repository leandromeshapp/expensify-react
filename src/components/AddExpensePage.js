import React from "react"
import ExpenseForm from "./ExpenseForm"
import { connect } from "react-redux"
import { startAddExpense } from "../actions/expenses"


export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        //props.dispatch(addExpense(expense))
        this.props.startAddExpense(expense)
        this.props.history.push("/")
    }

    render() {
        const {locale, dictionary} = this.props
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title"> {dictionary.pageAddExpense} </h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),

    locale: state.lang.locale,
    dictionary: state.lang.dictionary
})

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
})


export default connect(mapStateToProps, mapDispatchToProps)(AddExpensePage)