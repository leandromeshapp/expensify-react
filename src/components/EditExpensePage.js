import React from "react"
import { connect } from "react-redux" 
import ExpenseForm from "./ExpenseForm"
import { startEditExpense, startRemoveExpense } from "../actions/expenses"
import { ModalDialog } from './ModalDialog';
import Modal from 'react-modal';
//import Prompt from './Prompt';

export class EditExpensePage extends React.Component {
    state = {
        modalOpen: false
      };
    
    openModal = () => {
        console.log('open modal');
        Modal.setAppElement(document.getElementById('app'));
        this.setState(() => ({ modalOpen: true }));
    };
    
    closeModal = () => {
        this.setState(() => ({ modalOpen: false }));
    };
    
    
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push("/")
    }

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push("/")
    }

    render() {
        const {locale, dictionary} = this.props
        return (  
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title"> {dictionary.pageEditExpense} </h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        expense = {this.props.expense}
                        onSubmit = {this.onSubmit}
                    />

            </div>
                <button 
                    className="button button--secondary" 
                    onClick={this.openModal}> 
                    Remove Expense 
                </button>
                

                <ModalDialog
                    modalOpen={this.state.modalOpen}
                    handleModalCancel={this.closeModal}
                    contentLabel="Confirm Remove Expense"
                >

                <h3>Do You Really Want To Remove This Expense?</h3>
                <p>This cannot be undone!</p>
                <button className="button" onClick={this.closeModal}>
                    Cancel
                </button>
                <button
                    className="button button--destructive"
                    onClick={this.onRemove}>
                    Remove
                </button>
                </ModalDialog>
        </div>
        )
    }
}


const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),

    locale: state.lang.locale,
    dictionary: state.lang.dictionary
})


const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
})


export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)