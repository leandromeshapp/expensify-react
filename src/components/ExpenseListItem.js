import React from "react"
import { Link } from "react-router-dom"
import moment from "moment"
import numeral from "numeral"
//import locale from 'numeral/locales';

//numeral.locale('pt-pt');
const ExpenseListItem = ({ id, description, amount, createdAt, note }) => (
    <Link className="list-item" to={`/edit/${id}`}>
        <div>
            <h3 className="list-item__title"> 
                { description } 
            </h3>

            <span className="list-item__sub-title"> { moment(createdAt).format("D MMMM YYYY") } </span>
        </div>
        <span className="list-item__note">{note}</span>
        <h3 className="list-item__data"> { numeral(amount / 100).format("0,0.00$") }  </h3>
    </Link>
)

export default ExpenseListItem