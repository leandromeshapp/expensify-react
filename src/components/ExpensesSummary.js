import React from "react"
import { connect } from "react-redux"

import numeral from "numeral"
import locale from 'numeral/locales';

import selectExpenses from "../selectors/expenses"
import selectExpensesTotal from "../selectors/expenses-total"

import { Link } from "react-router-dom"
import { showAll } from "../actions/filters"

import replaceAll from '../utils/replaceAll';

export const ExpensesSummary = ({ expenses, expensesTotal, filters, visibleExpenseCount, visibleExpensesTotal, dictionary }) => {
    const expenseCount = expenses.length
    const filteredExpensesCount = expenseCount - visibleExpenseCount
    const visibleExpenseWord = visibleExpenseCount === 1 ? 'expense' : 'expenses'
    //const formattedVisibleExpensesTotal = numeral(visibleExpensesTotal / 100).format('$0,0.00')
    numeral.locale('pt-pt');
    const formattedVisibleExpensesTotal = numeral(visibleExpensesTotal / 100).format('0,0.00 $')
    // const formattedFilteredExpensesTotal = numeral((expensesTotal - visibleExpensesTotal) / 100).format('$0,0.00')
    const formattedFilteredExpensesTotal = numeral((expensesTotal - visibleExpensesTotal) / 100).format('0,0.00 $')
    
    const existingExpensesButFiltered = () => {
        if (expenseCount >= 1 && visibleExpenseCount < 1) {
          return true
        }
        return false
    };

    return (
        <div className="page-header">
            <div className="content-container">
            { expenseCount < 1 && 
                <h1 className="page-header__title"> {dictionary.summaryMessageNoExpenses} </h1>
            }

            { existingExpensesButFiltered() && 
                <h1 className="page-header__title"> {dictionary.summaryMessageNoMatch} </h1>
            }

            { visibleExpenseCount >= 1 && 
                <h1 className="page-header__title"
                    dangerouslySetInnerHTML={{
                        __html: replaceAll(dictionary.summaryMessageTitle, {
                            "{p1}": `<span>${expenseCount}</span>`,
                            "{p2}": expenseCount != 1 ? 's' : '',
                            "{p3}": `<span>${numeral(expensesTotal / 100).format('$0,0.00')}</span>`
                        })
                    }}
                >
                </h1>
            }
                
            <div className="page-header__action">
                <Link className="button button" to="/create"> {dictionary.pageAddExpense}</Link>
            </div>
            <br/>

            { expenseCount !== visibleExpenseCount &&
                <div className="header-filter-message"
                    dangerouslySetInnerHTML={{
                        __html: replaceAll(dictionary.summaryMessageSubitle, {
                            "{p1}": `<span>${filteredExpensesCount}</span>`,
             

                            "{p2}": filteredExpensesCount === 1 ? ' expense is not ' : ' expenses are not',
                                    

                            "{p3}": `<span>${formattedFilteredExpensesTotal}</span>`
                        })
                    }} 
                >
                </div>}
            </div>
        </div>
    )
} 


const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenses: state.expenses,
        expensesTotal: selectExpensesTotal(state.expenses),
        filters: state.filters,
        visibleExpenseCount: visibleExpenses.length,
        visibleExpensesTotal: selectExpensesTotal(visibleExpenses),

        locale: state.lang.locale,
        dictionary: state.lang.dictionary
    }
}

export default connect(mapStateToProps, {showAll})(ExpensesSummary)