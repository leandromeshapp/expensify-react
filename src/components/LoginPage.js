import React from 'react';
import { connect } from 'react-redux';
import { startLogin, startLoginEmail, createLoginEmail, startLoginGoogle, startLoginGithub, startLoginFacebook, startLoginTwitter } from '../actions/auth';
import { Redirect, App, Route, Switch} from "react-router-dom"
import { Button } from 'react-bootstrap'
import { Register } from './Register'
// export const LoginPage = ({ startLogin }) => (
import { setLanguage } from '../actions/lang'
let visibility = false

//export const LoginPage = ({ startLoginGoogle, startGithubLogin }) => (
export class LoginPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      emailError: '',
      passwordError: '',
      generalError: '',
    };
  }

  toggleVisibility = () => {
    visibility = !visibility

    this.forceUpdate()
  }

  changeLanguage = (e) => {
    const language = e.target.value
    console.log(language)
    this.props.setLanguage(language)
  };


  onEmailChange = (e) => {
    const email = e.target.value
    this.setState(() => ({ email }))
  }

  onPasswordChange = (e) => {
    const password = e.target.value
    this.setState(() => ({ password }))
  }

  onCreate () {
    <Redirect to = "/Register" />
  }

  onSubmit = (e) => {
    e.preventDefault()
    this.props.startLoginEmailProp(this.state.email, this.state.password).then((resErr) => {
      throw resErr
    }).catch((err) => {
      if (err.code === 'auth/invalid-email') {
        this.setState(() => ({ emailError: 'Invalid email format' }))
      }
      if (this.state.password.length < 6) {
        this.setState(() => ({ passwordError: 'Password must be at least six characterss' }))
      } else if (err.code !== 'auth/invalid-email' && this.state.password.length >= 6) {
        if (err.code === 'auth/wrong-password') {
          return console.log(err)
        }
        return this.setState(() => ({ generalError: 'Invalid login' }))
      }
      return undefined
    });
  }


  onCreateLoginEmail = () => {
    this.props.history.push("/Register")
    console.log("yau, fiz push para o register")
}

  visibility = !visibility
  render() {
    const {locale, dictionary} = this.props
    return (
    <div className='box-layout'>
      <div className='box-layout__box'>
        <h1 className='box-layout__title'>Expensify</h1>
        <p>It's time to get your expenses under control</p>
        {/* #dd4b39 */}

        <Button bsStyle="info" onClick={this.toggleVisibility}>
            {visibility ? 'Hide Social Login Options' : 'Show Social Login Options'}
        </Button>

        {visibility && (
          <div>
            <div className="buttonLogin">
              <button className='button button--with-icon' style={{"background":"#dd4b39"}} onClick={ this.props.startLoginGoogleProp }><i className="icon-prepend fa fa-google"/> Login with Google</button>
              <button className='button button--with-icon' style={{"background":"#4267b2"}} onClick={ this.props.startLoginFacebookProp }><i className="icon-prepend fa fa-facebook-f"/> Login with Facebook</button>
              <button className='button button--with-icon' style={{"background":"#24292e"}} onClick={ this.props.startLoginGithubProp }><i className="icon-prepend fa fa-github"/> Login with Github</button>
              <button className='button button--with-icon' style={{"background":"#1dcaff"}} onClick={ this.props.startLoginTwitterProp }><i className="icon-prepend fa fa-twitter"/> Login with Twitter</button>
            </div>
          </div>
        )}

        <br/>
        <br/>
        <h1>E-mail Login </h1>
        <form onSubmit={this.onSubmit} >
        <input
          className="form-control"
          type="text"
          placeholder="Email"
          autoFocus
          value={this.state.email}
          onChange={this.onEmailChange}
        />
        <br/>
        <input
          className="form-control"
          type="password"
          placeholder="Password"
          value={this.state.password}
          onChange={this.onPasswordChange}
        />

      <br/>
      <div className="botaoMargin">
        <Button bsStyle="primary" type="submit"> {dictionary.loginButton} </Button> 
        <br/>
        <Button bsStyle="danger" onClick={this.onCreateLoginEmail} > {dictionary.registerButton} </Button>

        <select className="select" value={locale} onChange={this.changeLanguage}>
          <option value="en">English</option>
          <option value="pt">Portuguese</option>
        </select>
      </div>
    </form>
  </div>
  </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startLoginGoogleProp: () => dispatch(startLoginGoogle()),
  startLoginGithubProp: () => dispatch(startLoginGithub()),
  startLoginFacebookProp: () => dispatch(startLoginFacebook()),
  startLoginTwitterProp: () => dispatch(startLoginTwitter()),
  startLogin: (provider) => dispatch(startLogin(provider)),
  startLoginEmailProp: (email, password) => dispatch(startLoginEmail(email, password)),
});


const mapStateToProps = (state) => ({
  locale: state.lang.locale,
  dictionary: state.lang.dictionary
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage)