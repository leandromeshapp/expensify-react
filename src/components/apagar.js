import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import { showAll } from "../actions/filters";
import selectExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";


export const ExpenseSummary = props => {
  const { expenses } = props;
  const expenseWord = expenses.length > 1 ? "expenses" : "expense";
  const formattedExpensesTotal = numeral(
    selectExpensesTotal(expenses) / 100
  ).format("$0,0.00");
  const filteredExpenseLength = props.filtered.length - expenses.length;
  const filteredWord = filteredExpenseLength > 1 ? "expenses" : "expense";

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          Viewing <span>{expenses.length} </span> {expenseWord} totalling
          <span> {formattedExpensesTotal} </span>
        </h1>

        {filteredExpenseLength !== 0 ? (
          <div className="hidden-title">
            <div className="hiddenMessage">
              Not showing{" "}
              <span className="length"> {filteredExpenseLength} </span>
              hidden {filteredWord} because of filters
            </div>
            <div>
              <button
                onClick={() => {
                  props.showAll();
                }}
                className="button button--showall"
              >
                View All
              </button>
            </div>
          </div>
        ) : (
          ""
        )}
        <div className="page-header__actions">
          <Link className="button" to="/create">
            Add Expense
          </Link>
        </div>
      </div>
    </div>
  );
};
const mapStateToProps = ({ expenses, filters }) => {

  return {
    expenses: selectExpenses(expenses, filters),
    filtered: expenses
  };
};
export default connect(mapStateToProps, { showAll })(ExpenseSummary);



// WEBPACK FOOTER //
// ./src/components/ExpenseSummary.js