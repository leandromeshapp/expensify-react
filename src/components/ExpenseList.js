import React from "react"
import { connect } from "react-redux"
import { Pagination } from 'react-bootstrap';

//Components
import ExpenseListItem from "./ExpenseListItem"
import selectExpenses from "../selectors/expenses"

import { orderExpenses  } from '../actions/expenses';


import ReactPaginate from "react-paginate"


export const ExpenseList = ({ expenses, dictionary }) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile"> {dictionary.tableExpenses}</div>
            <div className="show-for-desktop"> {dictionary.tableExpense}</div>
            <div className="show-for-desktop"> {dictionary.tableAmount}</div>
        </div>
        <div className="list-body">

        {/* //<button className="buttonRegister" onClick={ orderExpenses(expenses) }> Register </button> */}
            {
                expenses.length === 0 ? (
                    <div className="list-item--message">
                        <span> No expenses </span> 
                    </div>
                    
                ) : (
                    expenses.map((expense, index) => {
                        return <ExpenseListItem key={expense.id} {...expense} />
                    })
                )
            }

          {/* <ReactPaginate
        previousLabel={<i className="glyphicon glyphicon-chevron-left" aria-hidden="true"></i>}
        nextLabel={<i className="glyphicon glyphicon-chevron-right" aria-hidden="true"></i>}
        breakLabel={'...'}
        pageCount={expenses.last_page}
        marginPagesDisplayed={2}
        pageRangeDisplayed={5}
        initialPage={0}
        forcePage={2 - 1}
        disableInitialCallback={true}
        //onPageChange={this.paginationPressed}
        containerClassName={"pagination"}
        subContainerClassName={"pages pagination"}
        activeClassName={"active"}
      /> */}
        </div>
    </div>
)


const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters),

        orderExpenses: (expenses) => dispatch(orderExpenses( expenses )),

        dictionary: state.lang.dictionary
    }
}

export default connect(mapStateToProps)(ExpenseList)