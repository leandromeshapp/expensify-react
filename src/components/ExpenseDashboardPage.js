import React from "react"
import ExpenseList from "./ExpenseList"
import ExpenseListFilters from "./ExpenseListFilters"
import ExpensesSummary from "./ExpensesSummary"
import Footer from "./Footer"

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
        <Footer />
    </div>
)

export default ExpenseDashboardPage