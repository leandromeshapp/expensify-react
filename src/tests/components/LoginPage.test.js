import React from "react"
import { shallow } from "enzyme"
import { LoginPage } from "../../components/LoginPage"


// Login Component
test("Should render Login Page correctly", () => {
    const wrapper = shallow(<LoginPage />)

    expect(wrapper).toMatchSnapshot()
})


// Button Click call startLogin
test("Should call startLogin on button click", () => {
    const startLogin = jest.fn()
    const wrapper = shallow(<LoginPage startLogin={startLogin} />)
    
    wrapper.find("button").simulate("click")
    expect(startLogin).toHaveBeenCalled()
})