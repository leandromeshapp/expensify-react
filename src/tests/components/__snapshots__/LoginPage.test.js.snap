// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login Page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
       Expensify 
    </h1>
    <p>
       It's time to get your expenses under control. 
    </p>
    <button
      class="button"
    >
       Login  
    </button>
  </div>
</div>
`;
